### PR TITLE
feat(scanner): Initialize Scanner v4 health checks

### PR DESCRIPTION
**Pull request by RTann**
## Description

This initializes the Scanner v4 gRPC health checks as defined in https://github.com/stackrox/stackrox/pull/7630.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

Nah

### CODE REVIEW

**Review by dcaravel — COMMENTED** [review]
Whats the plan for authorization? 

Assuming the health check service will not be requiring mTLS so that k8s is able to call it, but the IndexerService will require it.

File: scanner/services/indexer.go [hunk 1]
```diff
@@ -18,10 +18,13 @@ import (
 	"github.com/stackrox/rox/pkg/grpc/authz/allow"
 	"github.com/stackrox/rox/scanner/indexer"
 	"google.golang.org/grpc"
+	grpchealth "google.golang.org/grpc/health/grpc_health_v1"
 )
 
 type indexerService struct {
+	grpchealth.UnimplementedHealthServer
```
Comment (jvdm) [inline scanner/services/indexer.go:25]: From the ADR I took you were going to create one Health service endpoint, which should perform different checks based on the mode Scanner is running on.

> Both modes of Scanner v4 execution, Indexer and Matcher, will expose this service (ScannerDB has no need for this). However, both modes will define "ready" (ie `status` of `SERVING`) differently.

If I understood correctly, what you are doing is augmenting each gRPC service (Indexer and Matcher) to offer the health RPCs. How the Kubelet would probe this when running Scanner in the Central Cluster? Two gRPC calls?

**Comment by RTann** [comment]
> Whats the plan for authorization?
> 
> Assuming the health check service will not be requiring mTLS so that k8s is able to call it, but the IndexerService will require it.

Correct, TLS is not currently supported by the probe (see [here](https://github.com/kubernetes/kubernetes/blob/v1.28.1/pkg/probe/grpc/grpc.go#L59)). Note it is possible to set different auth rules for different endpoints. For example, take a look at the [image service](https://github.com/stackrox/stackrox/blob/4.1.3/central/image/service/service_impl.go#L53). So, we can set some auth rules for the other endpoints, and not for this one

**Comment by RTann** [comment]
superseded by https://github.com/stackrox/stackrox/pull/9531

**Pull request closed without merging**